atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/NodesDownTestSetup.java

```diff
@@ -29,32 +31,50 @@
 import java.time.Duration;
 import java.util.List;
 import org.awaitility.Awaitility;
-import org.junit.AfterClass;
-import org.junit.ClassRule;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 
-public abstract class NodesDownTestSetup {
+public class NodesDownTestSetup implements BeforeAllCallback, ExtensionContext.Store.CloseableResource {
+    private static boolean started = false;
     private static final int CASSANDRA_THRIFT_PORT = 9160;
     private static final CassandraKeyValueServiceConfig CONFIG = ImmutableCassandraKeyValueServiceConfig.copyOf(
                     ThreeNodeCassandraCluster.KVS_CONFIG)
             .withSchemaMutationTimeoutMillis(3_000);
     private static final Refreshable<CassandraKeyValueServiceRuntimeConfig> RUNTIME_CONFIG =
             ThreeNodeCassandraCluster.KVS_RUNTIME_CONFIG;
 
-    @ClassRule
-    public static final Containers CONTAINERS =
-            new Containers(NodesDownTestSetup.class).with(new ThreeNodeCassandraCluster());
+    public static final ContainersV2 CONTAINERS =
+            new ContainersV2(NodesDownTestSetup.class).with(new ThreeNodeCassandraCluster());
 
-    @AfterClass
-    public static void closeKvs() {
-        AbstractDegradedClusterTest.closeAll();
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
+        if (!started) {
+            CONTAINERS.beforeAll(context);
+            started = true;
+            for (Class<?> test : List.of(
+                    OneNodeDownAvailabilityTest.class,
+                    OneNodeDownDeleteTest.class,
+                    OneNodeDownGetTest.class,
+                    OneNodeDownMetadataTest.class,
+                    OneNodeDownPutTest.class,
+                    OneNodeDownTableManipulationTest.class,
+                    ThreeNodesDownAvailabilityTest.class,
+                    TwoNodesDownAvailabilityTest.class,
+                    TwoNodesDownGetTest.class,
+                    TwoNodesDownMetadataTest.class,
+                    TwoNodesDownPutTest.class,
+                    TwoNodesDownTableManipulationTest.class)) {
+                test.getMethod("initialize", CassandraKeyValueService.class)
+                        .invoke(test.getDeclaredConstructor().newInstance(), createKvs(test));
+            }
+            context.getRoot().getStore(GLOBAL).put("any unique name", this);
+        }
```

Comment:
It's hard to explain. Before killing any node, we need to initialize KVS for all test classes. Otherwise, KVS initialization fails since some nodes are down. This code piece should run only once because if it runs after some test classes have executed, some nodes may be down. This is from the old implementation; however, we need to hard-code the class names, which is unfortunate.